src/TemplateHelpers.hs

```diff
@@ -125,7 +128,18 @@ renderReadme = \case
 renderHtmlDocs :: D.VerifiedPackage -> Text -> Handler (Maybe Html)
 renderHtmlDocs pkg mnString = do
   htmlRenderContext <- getHtmlRenderContext
-  let docsOutput = packageAsHtml (htmlRenderContext pkg) pkg
+
+  depHtmlRenderContexts <- do
```

Comment:
Hm, how about we just move the entire `deplHtmlRenderContexts` into a where clause? This way you get type annotations to indicate how far the function has been applied and all complexity is localized. What do you think?